compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/SchemaGenerator.java

```diff
@@ -480,11 +515,21 @@ private Field getField(RecordFieldSymbol recordFieldSymbol, String description)
         String name = recordFieldSymbol.getName().get();
         Field field = new Field(name, description);
         TypeSymbol typeSymbol = recordFieldSymbol.typeDescriptor();
-        Type type;
+        Type type = null;
         if (typeSymbol.typeKind() == TypeDescKind.MAP) {
             type = getType(((MapTypeSymbol) typeSymbol).typeParam());
             Type argType = getWrapperType(addType(ScalarType.STRING), TypeKind.NON_NULL);
             field.addArg(new InputValue(MAP_KEY_ARGUMENT_NAME, argType, MAP_KEY_ARGUMENT_DESCRIPTION, null));
+        } else if (!recordFieldSymbol.annotations().isEmpty()) {
+            for (AnnotationSymbol annotationSymbol : recordFieldSymbol.annotations()) {
+                if (annotationSymbol.getName().isPresent() && annotationSymbol.getName().get().equals(ID_ANNOT_NAME)) {
+                    type = getTypeForID(recordFieldSymbol.typeDescriptor());
```

Comment:
I added a separate method because of this reason.
To check if it is ID in `getInputType` I will have to pass something like a boolean, and for other input types also this will be checked all the time. 
eg: 
if (isIdType) {
   return idtype;
} else {
   process other types
   // other types check the type from the  `typeSymbol.typeKind()` which can't be used to check if the type is ID :/ 
}
or whole annotation processing will have to brought into `getInputType` method, which would mess it up i think. it looks nice and neat now, didn't want to eat it. 

WDYT

Reply:
Legit reason.